docs/features/dpdk/linux-networking.md

```diff
@@ -0,0 +1,431 @@
+  LINK port is bound to the DPDK target.
+- VLAN 1, VLAN 2, .... VLAN N are created using Linux commands, and are on
+  top of a TAP port. The number of VLAN ports should be equal to the number
+  of VM's that are spawned.
+- br-int, VxLAN0 ports are created using the ovs-vsctl command provided by OvS.
+  All VLAN ports are attached to br-int using ovs-vsctl command.
+
+System under test will have above topology running the networking recipe.
+Link Partner can have either the networking recipe or legacy OVS or kernel
+VxLAN. Note the [Limitations](#limitations) below before setting up the topology.
+
+## Create P4 artifacts
+
+1. Install the [p4c compiler](https://github.com/p4lang/p4c)
+   from the p4lang/p4c repository and follow the instructions in the
+   [readme file](https://github.com/p4lang/p4c/blob/main/README.md).
+
+2. Set the environment variable OUTPUT_DIR to the location where artifacts
+   should be generated and p4 files are available. For example:
+
+   ```bash
+   export OUTPUT_DIR=$IPDK_RECIPE/p4src/linux_networking
+   ```
+
+3. Compile the P4 program using p4c-dpdk:
+
+   ```bash
+   p4c-dpdk --arch pna --target dpdk \
+       --p4runtime-files $OUTPUT_DIR/p4Info.txt \
+       --bf-rt-schema $OUTPUT_DIR/bf-rt.json \
+       --context $OUTPUT_DIR/context.json \
+       -o $OUTPUT_DIR/linux_networking.spec \
+       $OUTPUT_DIR/linux_networking.p4
+   ```
+
+4. Modify the sample configuration file
+   ($IPDK_RECIPE/p4src/linux_networking/dpdk/lnw.conf)
+   to specify the absolute paths of the artifacts (json and spec files).
+
+5. Generate binary image using tdi-pipeline builder:
+
+   ```bash
+   tdi_pipeline_builder --p4c_conf_file=lnw.conf \
+       --bf_pipeline_config_binary_file=lnw.pb.bin
+   ```
+
+## Limitations
+
+Current SAI enablement for the networking recipe has following limitations:
+
+- All VHOST user ports need to be configured first and only then TAP ports/physical ports
+- TAP port created for the corresponding link port should be created using "gnmi-ctl control port creation got the link port"
+  e.g., gnmi-ctl set "device:physical-device,name:PORT0,pipeline-name:pipe,mempool-name:MEMPOOL0,control-port:TAP1,mtu:1500,pci-bdf:0000:18:00.0,packet-dir=network,port-type:link"
+- All VLAN interfaces created on top of TAP ports should always be in lowercase
+  format "vlan+vlan_id" (e.g, vlan1, vlan2, vlan3, ... vlan4094).
+- br-int port, vxlan0 port and adding vlan ports to br-int need to be done after loading the pipeline.
+- VxLAN destination port should always be standard port (4789). (limitation by p4 parser)
+- Only VNI 0 is supported.
+- We are not supporting any ofproto rules which would not allow for FDB learning on OVS.
+
+## Steps to create the topology
+
+The [gnmi-ctl](/clients/gnmi-ctl) and [p4rt-ctl](/clients/gnmi-ctl) utilities
+used below can be found in $IPDK_RECIPE/install/bin. They should be run with
+`sudo`.
+
+## 1) Bind physical port to IO driver
```

Comment:
Yes, I've backtracked somewhat on the question of numbering paragraphs.

We definitely do not want standard hierarchical numbering. That's appropriate for a reference document, not for a page or chapter in a manual.

But this page has a _loooong_ list of steps, and I'm concerned about the user losing their place in the document.

One of the documents I was read earlier this week -- I think it was part of the ReadTheDocs developer documentation -- said something to the effect that really long pages should generally be broken up into smaller pages. (Rather like really large functions or modules...)

Logistically, you'd either have a much more compact list, each of whose items has a link to a page that documents the item, or you'd have a series of chained pages, each with links to the preceding or following page. (Basically, a doubly-linked list.)

Some of these pages might well be shared -- one on starting OvS, one on starting infrap4d, for example.

If the Networking Recipe variants are all variations of the same document, we might very well want  whole sections that are common.